### PR TITLE
[Bug] Fix eggs having exploitable RNG

### PR DESCRIPTION
**Pull request by PigeonBar**
## What are the changes the user will see?

Egg RNG manipulation will no longer work.

## Why am I making these changes?

A temporary fix for #3607 before Xavion's planned RNG rework.

## What are the changes from a developer perspective?

Implements 2 fixes at the same time:

1. Properly resets RNG upon saving and quitting into the title screen. This protects against RNG manipulating anything accessible from the title screen, including starter eggs.
    - Also fixes some unit tests that have a new random chance to fail due to these RNG resets.
2. In `egg.ts`, uses `scene.executeWithSeedOffset()` to make egg RNG unseeded and unpredictable. This protects against RNG manipulation for all eggs, including candy eggs and gacha eggs.

## How to test the changes?

Here were the steps for reproducing the original bug:

1. Load a session (after saving and quitting at least once).
2. Save and quit (NOT refresh).
3. Navigate to the starter select screen.
4. Buy a single starter egg using candies.
5. Repeat from Step 1 for each egg.

If the fix works properly, then the eggs should have different egg move slots, ability slots, shininess, etc..

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR? (not until Xavion's planned RNG rework)
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~~[ ] Are the changes visual?~~
  - ~~[ ] Have I provided screenshots/videos of the changes?~~

### CODE REVIEW

**Comment by RalseiGithub** [comment]
Feel like this would make speedrun S a lot easier if they exist

**Comment by PigeonBar** [comment]
Alright, I think `wind_rider.test.ts` turns out to be one of those "random chance to fail" tests. I'll look into this some more later.

**Review by DayKev — APPROVED** [review]

**Comment by DayKev** [comment]
Lemme break out those test fixes into their own PR.

**Comment by DayKev** [comment]
#3968 should resolve the test issues

File: src/test/abilities/wind_rider.test.ts [hunk 1]
```diff
@@ -88,6 +88,7 @@ describe("Abilities - Wind Rider", () => {
     await game.classicMode.startBattle([Species.SHIFTRY]);
     const magikarp = game.scene.getEnemyPokemon()!;
     const shiftry = game.scene.getPlayerPokemon()!;
+    expect(shiftry.hasAbility(Abilities.WIND_RIDER)).toBe(true);
```
Comment (DayKev) [inline src/test/abilities/wind_rider.test.ts:91]:
```suggestion
```

**Comment by frutescens** [comment]
Ugh. I thought I uploaded my video from my test here.


https://github.com/user-attachments/assets/b870f392-7d68-4429-8feb-a7494312856b

**Review by frutescens — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**